### PR TITLE
kmsxx: 2020-08-04 -> 2021-07-26 unbreak

### PR DESCRIPTION
**Pull request by Artturin**
###### Motivation for this change
Supersedes  #111976

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 132950` at fb9801b2 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/4b050db4a230ba09b054cf877fe6ca8f">python38Packages.kmsxx</a></li>
    <li><a href="https://gist.github.com/8c2612f96afbe2b267e21df982892498">python39Packages.kmsxx</a></li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>kmsxx</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/kmsxx/default.nix:39:5:
```
   |
39 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/libraries/kmsxx/default.nix [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
-    sha256 = "0xz4m9bk0naawxwpx5cy1j3cm6c8c9m5y551csk88y88x1g0z0xh";
+    rev = "54f591ec0de61dd192baf781c9b2ec87d5b461f7";
+    sha256 = "sha256-j+20WY4a2iTKZnYjXhxbNnZZ53K3dHpDMTp+ZulS+7c=";
   };
 
-  cmakeFlags = lib.optional (!withPython) "-DKMSXX_ENABLE_PYTHON=OFF";
+  # Didn't detect pybind11 without cmake
+  nativeBuildInputs = [ meson ninja pkg-config ] ++ lib.optionals withPython [ cmake ];
+  buildInputs = [ libdrm fmt libevdev ]
+    ++ lib.optionals withPython ( with python3Packages; [ python pybind11 ]);
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kmsxx/default.nix:30]:
```suggestion
    ++ lib.optionals withPython (with python3Packages; [ python pybind11 ]);
```

File: pkgs/development/libraries/kmsxx/default.nix [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
     fetchSubmodules = true;
-    rev = "38bee3092f2d477f1baebfcae464f888d3d04bbe";
-    sha256 = "0xz4m9bk0naawxwpx5cy1j3cm6c8c9m5y551csk88y88x1g0z0xh";
+    rev = "54f591ec0de61dd192baf781c9b2ec87d5b461f7";
+    sha256 = "sha256-j+20WY4a2iTKZnYjXhxbNnZZ53K3dHpDMTp+ZulS+7c=";
   };
 
-  cmakeFlags = lib.optional (!withPython) "-DKMSXX_ENABLE_PYTHON=OFF";
+  # Didn't detect pybind11 without cmake
+  nativeBuildInputs = [ meson ninja pkg-config ] ++ lib.optionals withPython [ cmake ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kmsxx/default.nix:28]: Can we replace pkg-config all the time with cmake?

Comment (Artturin) [reply pkgs/development/libraries/kmsxx/default.nix:28]:
```
kmsxx> Did not find pkg-config by name 'pkg-config'
kmsxx> Found Pkg-config: NO
kmsxx> Found CMake: /nix/store/p1lk3bb1wzyn5v3nw8nyxrc44iy60vwk-cmake-3.19.7/bin/cmake (3.19.7)
kmsxx> WARNING: CMake: Dependency -Wl,--as-needed for fmt target fmt::fmt was not found
kmsxx> Run-time dependency fmt found: YES 7.1.3
kmsxx> Run-time dependency libdrm_omap found: NO (tried cmake)
kmsxx> meson.build:37:0: ERROR: Pkg-config binary for machine MachineChoice.HOST not found. Giving up.
```

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 132950 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>kmsxx</li>    <li>python38Packages.kmsxx (python39Packages.kmsxx)</li>  </ul></details>

**Pull request merged**